app/javascript/components/subject/CaseStatus.js

```diff
@@ -23,58 +23,47 @@ class CaseStatus extends React.Component {
       loading: false,
     };
     this.origState = Object.assign({}, this.state);
-    this.toggleCaseStatusModal = this.toggleCaseStatusModal.bind(this);
-    this.submit = this.submit.bind(this);
-    this.handleChange = this.handleChange.bind(this);
   }
 
-  handleChange(event) {
+  handleChange = event => {
     event.persist();
-
     const value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
-    const confirmedOrProbable = value === 'Confirmed' || value === 'Probable';
-    const hideModal = this.state.isolation && confirmedOrProbable;
 
-    this.setState({ [event.target.id]: value, showCaseStatusModal: !hideModal, confirmedOrProbable }, () => {
-      // specific case where case status is just changed with no modal
-      if (hideModal) {
-        this.submit();
-      }
+    this.setState({ [event.target.id]: value }, () => {
+      if (event.target.id === 'case_status') {
+        const confirmedOrProbable = value === 'Confirmed' || value === 'Probable';
+        const hideModal = this.state.isolation && confirmedOrProbable;
 
-      // all other cases
-      if (event.target.id === 'confirmed') {
+        this.setState({ showCaseStatusModal: !hideModal, confirmedOrProbable }, () => {
+          // specific case where case status is just changed with no modal
+          if (hideModal) {
+            this.submit();
+          }
+          if (!confirmedOrProbable) {
+            this.setState({ isolation: false, public_health_action: 'None' });
+          }
+        });
+      } else if (event.target.id === 'confirmed') {
```

Comment:
Looking through this, I think it's a bit confusing that this is called `confirmed` given that that is also a case status option and that's not exactly what this is checking. I would suggest renaming to avoid confusion/bugs.

Reply:
I agree - changed it